lib/xlibc/src/threads/tss_create.c

```diff
@@ -0,0 +1,37 @@
+#include <threads.h>
+#include <assert.h>
+#include <_tss.h>
+
+mtx_t tss_lock;
+// Note: This uses range-based initialization, which is a GNU C extension
```

Comment:
Is this really necessary?

Reply:
Every element of these arrays needs to be initialized before using any TSS-touching code, so my options were:
1. Introduce an initialization function which gets run before creating the thread for main() (ugly imho and would shift initialization to runtime)
2. Use humongous amounts of dirty macro magic, which expands to a proper initializer list
3. Use the GNU extension

I went with 3 because it seemed the least messy to me. Also I would have expected C to have this as a properly standardized feature already.